src/Microsoft.ML.Core/EntryPoints/ComponentFactory.cs

```diff
@@ -29,6 +29,25 @@ public interface IComponentFactory<out TComponent>: IComponentFactory
         TComponent CreateComponent(IHostEnvironment env);
     }
 
+    /// <summary>
+    /// A class for creating a component with no extra parameters (other than an <see cref="IHostEnvironment"/>)
+    /// that simply wraps a delegate which creates the component.
+    /// </summary>
+    public class SimpleComponentFactory<TComponent> : IComponentFactory<TComponent>
```

Comment:
>public class SimpleComponentFactory [](start = 4, length = 35)

I'm somewhat reluctant to expose more public types in addition to `IComponentFactory` -- maybe if we just had something simple like a static creation class that returned `IComponentFactory`, with things like:

```csharp
public static class ComponentFactoryUtils {
    public static IComponentFactory<TComponent> CreateFromFunction(Func<IHostEnvironment, TComponent> factory) { ... }
    public static IComponentFactory<TArg1, TComponent> CreateFromFunction(Func<IHostEnvironment, TArg1, TComponent> factory) { ... }
    public static IComponentFactory<TArg1, TArg2, TComponent> CreateFromFunction(Func<IHostEnvironment, TArg1, TArg2, TComponent> factory) { ... }
}
```

Then we could hide the actual implementations of these.

Reply:
This is a good idea. It also simplifies the call sites, in some cases, since the types can be inferred.  Fixed.

---
In reply to: [210434551](https://github.com/dotnet/machinelearning/pull/681#discussion_r210434551) [](ancestors = 210434551)